tests/certification/pubsub/azure/eventhubs/eventhubs_test.go

```diff
@@ -0,0 +1,301 @@
+	// addExpectedMessagesforIot := func(messageWatchers *watcher.Watcher) flow.Runnable {
+	// 	return func(ctx flow.Context) error {
+	// 		messages := make([]string, numMessages)
+	// 		for i := range messages {
+	// 			messages[i] = fmt.Sprintf("testmessageForEventHubCertificationTest #%v", i+1)
+	// 		}
+	// 		messageWatchers.ExpectStrings(messages...)
+	// 		return nil
+	// 	}
+	// }
+
+	flow.New(t, "eventhubs certification").
+
+		// Test : single publisher, multiple subscriber with their own consumerID
+		// Run subscriberApplication app1
+		Step(app.Run(appID1, fmt.Sprintf(":%d", appPort),
+			subscriberApplication(appID1, topicName1, consumerGroup1))).
+
+		// Run the Dapr sidecar with the eventhubs component 1.
+		Step(sidecar.Run(sidecarName1,
+			embedded.WithComponentsPath("./components/consumer1"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+
+		// Run subscriberApplication app2
+		Step(app.Run(appID2, fmt.Sprintf(":%d", appPort+portOffset),
+			subscriberApplication(appID2, topicName1, consumerGroup2))).
+
+		// Run the Dapr sidecar with the component 2.
+		Step(sidecar.Run(sidecarName2,
+			embedded.WithComponentsPath("./components/consumer2"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+
+		// Run dummy app
+		Step(app.Run(dummyAppID, fmt.Sprintf(":%d", appPort+portOffset*10),
+			dummyApp(dummyAppID))).
+		// Run the dummy Dapr sidecar with the dummyconsumer component, to be used for publishing messages to a topic with no subscribers (unused)
+		Step(sidecar.Run(sidecarDummyName,
+			embedded.WithComponentsPath("./components/dummyconsumer"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset*10),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset*10),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset*10),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset*10),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+		Step("publish messages to topic1", publishMessages(nil, sidecarName1, topicName1, consumerGroup1, consumerGroup2)).
+		// Start a sidecar with a component refering eh connection string of unusedTopic
+		Step("publish messages to unUsedTopic", publishMessages(nil, sidecarDummyName, unUsedTopic)).
+		Step("verify if app1 has recevied messages published to topic1", assertMessages(10*time.Second, consumerGroup1)).
```

Comment:
is negative assertion possible ? 
To verify it has not received any messages from the unused topic?

Reply:
Package `watcher` being used here (also in kafka) ignores the messages which are not part of expected strings. 

One possible solution i can think of is exposing actual & expected arrays data types via changing the scope or a different function.

Once we have access we can run set operations to achieve our goal.

Reply:
To be tracked as part of : https://github.com/dapr/components-contrib/issues/1663